apps/omg/lib/omg/output.ex

```diff
@@ -37,9 +37,13 @@ defmodule OMG.Output do
   Reconstructs the structure from a list of RLP items
   """
   def reconstruct([raw_type, [_owner, _currency, _amount]] = rlp_data) when is_binary(raw_type) do
-    with :ok <- validate_data(rlp_data),
-         {:ok, utxo} <- ExPlasma.Utxo.new(rlp_data),
-         do: %__MODULE__{output_type: utxo.output_type, owner: utxo.owner, currency: utxo.currency, amount: utxo.amount}
+    with {:ok, type, owner, currency, amont} <- clean_and_validate_data(rlp_data),
+         do: %__MODULE__{
+           output_type: type,
+           owner: owner,
+           currency: currency,
+           amount: amont
+         }
```

Comment:
I don't know but I don't like the `with` clause when there's only one function call and no specs. what is this supposed to return, except the struct?
if nothing can go wrong, why the defensive coding?
if there are error tuples, do we know what we can expect to see?

Reply:
Right. I added specs. They are very basic, but from what I've seen how it is being used, we really do not care about particular error type, in prod code we map all returned errors to `{:error, :malformed_outputs}`.
I can improve on that in a separate PR if needed. But let's leave scope of this PR to just using ExPlasma from hex.